public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx

```diff
@@ -26,34 +28,82 @@ export const PromExploreQueryEditor: FC<Props> = (props: Props) => {
     }
   }
 
+  function onQueryTypeChange(value: string) {
+    const { query, onChange } = props;
+    let nextQuery;
+    if (value === 'instant') {
+      nextQuery = { ...query, instant: true, runAll: false };
+    } else if (value === 'range') {
+      nextQuery = { ...query, instant: false, runAll: false };
+    } else {
+      nextQuery = { ...query, instant: true, runAll: true };
```

Comment:
this mapping seems a bit confusing to me. Why don't we have just:
```
all => {range: true, instant: true}
instant => {range: false, instant: true}
range => {range: true, instant: false}
```
?

Reply:
>So did we decide to not have collapsing at all? Not even on the ui level that would not affect the queries?

That was my understanding. I even double-check at the call, but maybe I misunderstood. +cc @davkal 

>this mapping seems a bit confusing to me. Why don't we have just...

Yeah, that was my first impulse as well, but then it got a little bit more complicated with visualisations (see more detailed explanation below) and (at least to me), this option with runAll seemed cleaner. But let me know what do you think. 

__________________________
How is query + query results processed:
1. `prepareTargets` function where we process and prepare targets. 
2. `createQuery` where we are creating each query, so when we have range + instant, we create 2 separate queries.  And we would rewrite `instant = false` for range query (as we are creating range query) and `range = false` for instant query (as it is instant query). 
3. `query (both exploreQuery and panelQuery)` run query - if `query.instant === true` run instant query, otherwise run range query
4. `process result` to process results and determine visualisation. And here comes the tricky part with 2 possible solutions that I could come up with:

- To make sure we are displaying correct visualisation such as in this `preferredVisualisationType: target.instant ? 'table' : target.runAll ? 'graph' : undefined,`, we would need to keep the information about if both queries (instant + range) where run, or if only 1 was run. Which means, that in `createQuery` we couldn't rewrite `instant = false` for range query and `range = false` for instant query and we would have to keep the original value (which is for me more confusing - it is instant query, but why does it have `range = true` in params). And then in `query` we would be relying only on `instant` and ignore `range` and used `range` only in `process result` to determine visualisation. 

- In `prepareTargets` we would add another key-value pair for `target` -> something like `runInstantAndRange = true/false` and then we would use it in the same was as `runAll` to determine visualisation. I like this option as well, but using `runAll` at the beginning seemed like less-code/more straight forward option and I have decided to go with that. But if you think this would be better (and as I was writing this, I started to like this option a little but more than my implementation) , I can refactor it. :)

Reply:
Based on our chat, I have refactored the solution in https://github.com/grafana/grafana/pull/27026/commits/5039f2c94c92d1e4df38aa77a967d605384fc67e to following:
- runAll removed and replaced with range 
- created variable mixedQueries to determine the visualisation